api/src/main/java/org/openmrs/config/OpenmrsServiceConfig.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.config;
+
+import org.hibernate.SessionFactory;
+import org.openmrs.aop.AuthorizationAdvice;
+import org.openmrs.aop.LoggingAdvice;
+import org.openmrs.aop.RequiredDataAdvice;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.db.hibernate.HibernatePatientDAO;
+import org.openmrs.api.db.hibernate.search.session.SearchSessionFactory;
+import org.openmrs.api.impl.PatientServiceImpl;
+import org.openmrs.api.db.PatientDAO;
+import org.openmrs.patient.IdentifierValidator;
+import org.openmrs.patient.impl.LuhnIdentifierValidator;
+import org.openmrs.patient.impl.VerhoeffIdentifierValidator;
+import org.springframework.cache.interceptor.CacheInterceptor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.hibernate5.HibernateTransactionManager;
+import org.springframework.transaction.annotation.AnnotationTransactionAttributeSource;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.transaction.interceptor.TransactionProxyFactoryBean;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+@Configuration
+@EnableTransactionManagement
+public class OpenmrsServiceConfig {
```

Comment:
@rkorytkowski  are you suggesting something like this, or something else?

Reply:
okay thank you, let me take a look and make the changes

Reply:
I would use `ClassFilter` for our advisors.